lib/stanford-mods/date_parsing.rb

```diff
@@ -0,0 +1,67 @@
+# Parsing date strings
+# These methods may be used by searchworks.rb file or by downstream apps
+module Stanford
+  module Mods
+    class DateParsing
+
+      # @param [String] date_str String containing four digit year (we hope)
+      # @return [String, nil] 4 digit year (e.g. 1865, 0950) if date_str has yyyy, nil otherwise
+      def self.sortable_year_from_date_str(date_str)
+        matches = date_str.match(/\d{4}/) if date_str
+        return matches.to_s if matches
+      end
+
+      # get year if we have a x/x/yy or x-x-yy pattern
+      #   note that these are the only 2 digit year patterns found in our actual date strings in MODS records
+      # @param [String] date_str String containing x/x/yy or x-x-yy date pattern
+      # @return [String, nil] 4 digit year (e.g. 1865, 0950) if date_str matches pattern, nil otherwise
+      def self.sortable_year_from_yy(date_str)
+        slash_matches = date_str.match(/\d{1,2}\/\d{1,2}\/\d{2}/) if date_str
+        if slash_matches
+          date_obj = Date.strptime(date_str, '%m/%d/%y')
+        else
+          hyphen_matches = date_str.match(/\d{1,2}-\d{1,2}-\d{2}/) if date_str
+          date_obj = Date.strptime(date_str, '%m-%d-%y') if hyphen_matches
+        end
+        if date_obj && date_obj > Date.today
+          date_obj = Date.new(date_obj.year - 100, date_obj.month, date_obj.mday)
```

Comment:
I guess we're assuming two digit years are `19xx` dates? Maybe that's worth a comment?